# ã‚¹ã‚¿ãƒƒãƒ•ï¼ˆStaffï¼›åˆ¶ä½œè€…ï¼‰
#	ã‚­ãƒ¼ã‚³ãƒ³ãƒ•ã‚£ã‚°ã®ä½¿ã„æ–¹ã®å®Ÿä¾‹ã§ã™ã€‚ã“ã®ãƒãƒ¼ãƒ‰è‡ªèº«ã¯ã‚­ãƒ¼ã‚³ãƒ³ãƒ•ã‚£ã‚°æœ¬ä½“ã§ã¯ã‚ã‚Šã¾ã›ã‚“
#
extends Node2D


# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
# ãƒãƒ¼ãƒ‰ãƒ»ãƒ‘ã‚¹é–¢é€£
# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼


# ãƒ‡ã‚£ãƒ¬ã‚¯ã‚¿ãƒ¼ãƒ»ãƒãƒ–å–å¾—
func monkey():
	return $"ğŸµMonkey"


# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
# ãã®ä»–
# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼


# Called when the node enters the scene tree for the first time.
func _ready():

	# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
	# éè¡¨ç¤º
	# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
	#
	# é–‹ç™ºä¸­ã«ã„ã˜ã£ãŸã‚‚ã®ãŒæ®‹ã£ã¦ã‚‹ã‹ã‚‚ã—ã‚Œãªã„ã‹ã‚‰ã€æƒé™¤
	
	# ã‚°ãƒªãƒƒãƒ‰ã¯éš ã™
	self.monkey().grid().hide()

	# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
	# è¡¨ç¤º
	# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼

	# ãƒ†ãƒ­ãƒƒãƒ—ãƒ»ã‚³ãƒ¼ãƒ‡ã‚£ãƒãƒ¼ã‚¿ãƒ¼è¡¨ç¤º
	self.monkey().telop_coordinator().show()

	# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
	# åˆæœŸåŒ–
	# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
	self.monkey().programmer().key_config_node().ready_in_staff()
	self.monkey().programmer().key_config_node().entry()


# Called every frame. 'delta' is the elapsed time since the previous frame.
func _process(delta):
	# ã‚­ãƒ¼ã‚³ãƒ³ãƒ•ã‚£ã‚°ã§å¿…è¦ã§ã™
	self.monkey().programmer().key_config_node().on_process(delta)


func _unhandled_input(event):
	# ã‚­ãƒ¼ã‚³ãƒ³ãƒ•ã‚£ã‚°ã§å¿…è¦ã§ã™
	self.monkey().programmer().key_config_node().on_unhandled_input(event)


# ã‚­ãƒ¼ã‚³ãƒ³ãƒ•ã‚£ã‚°é–‹å§‹æ™‚
func on_key_config_entered():
	# èƒŒæ™¯
	self.monkey().illustrator().visible = true
	self.monkey().programmer().owner_node().images.find_node("ğŸ—»å´å·é§…å‰").visible = true


# ã‚­ãƒ¼ã‚³ãƒ³ãƒ•ã‚£ã‚°å®Œäº†æ™‚
func on_key_config_exited():
	print("ï¼»ã‚­ãƒ¼ã‚³ãƒ³ãƒ•ã‚£ã‚°ï¼½ã€€å®Œäº†")

	# èƒŒæ™¯
	self.monkey().programmer().owner_node().images.find_node("ğŸ—»å´å·é§…å‰").visible = false
